src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -861,9 +861,17 @@ public void CloseDocumentCore(DocumentId documentId)
 
         protected override void ApplyDocumentTextChanged(DocumentId documentId, SourceText newText)
         {
-            EnsureEditableDocuments(documentId);
-            var hostDocument = GetHostDocument(documentId);
-            hostDocument.UpdateText(newText);
+            var filename = GetFilePath(documentId);
+            // When making project wide changes(i.e. change framework version),
+            // the TemporaryGeneratedFile_xxxx files under obj\Debug get removed and generated again.
```

Comment:
Put another way: this might fix it, but silently failing to apply a text change to something that moved seems we have a synchronization problem somewhere more fundamental.

Reply:
reverted. Changed the way of comparing DocumentState instead :)